trevorwinser/Infinite-Loopers
# Login Authorization on All Webpages

Completed #169 

Test code using the following steps:

1. Navigate to _server_ directory using `cd server`
2. Setup your virtual environment.
3. Run `pip3 install -r ../requirements.txt`
4. Run `flask run`

This should open up a flask app, and navigating to http://127.0.0.1:5000 should bring you to the login page.

### Testing the Login Authorization

5. Try navigating to a webpage using its address in your browser's address bar.
6. You should automatically be redirected to the login page.

### Testing Admin authorization

7. Try logging using the user credentials:

Username:  User1
Password: Password1

8. And then trying to navigate to an admin page, such as http://127.0.0.1:5000/admin or http://127.0.0.1:5000/users
9. You should be redirected back to the user-dashboard.

## What's New?

* Users that aren't logged in will not be able to access other pages, and will be redirected back to the login page when they try to do so.
* Non-Admins cannot access admin pages even if they are logged in, and will be redirected to the user dashboard when they  try to do so.
